### ISSUE TITLE
pip-compile requirements.in: Permission Error

### ISSUE THREAD

**Issue by MrLeeh** [nvie/pip-tools#187]
I tried using
```bash
pip-compile requirements.in
```
but get the following error:
```bash
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\piptools\io.py", line 92, in setup
    overwrite=self.overwrite_part)
  File "C:\Python34\lib\site-packages\piptools\io.py", line 14, in _atomic_rename
    os.rename(path, new_path)
PermissionError: [WinError 32] Der Prozess kann nicht auf die Datei zugreifen, da sie von einem ande
ren Prozess verwendet wird: 'c:\\Users\\Lehmann\\data\\python\\python34\\moistcalc\\tmpmoibd2g8' ->
'requirements.txt.part'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python34\lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python34\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python34\Scripts\pip-compile.exe\__main__.py", line 9, in <module>
  File "C:\Python34\lib\site-packages\click\core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "C:\Python34\lib\site-packages\click\core.py", line 644, in main
    rv = self.invoke(ctx)
  File "C:\Python34\lib\site-packages\click\core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Python34\lib\site-packages\click\core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "C:\Python34\lib\site-packages\piptools\scripts\compile.py", line 119, in cli
    primary_packages={ireq.req.key for ireq in constraints})
  File "C:\Python34\lib\site-packages\piptools\writer.py", line 80, in write
    f = stack.enter_context(AtomicSaver(self.dst_file))
  File "C:\Python34\lib\contextlib.py", line 283, in enter_context
    result = _cm_type.__enter__(cm)
  File "C:\Python34\lib\site-packages\piptools\io.py", line 101, in __enter__
    self.setup()
  File "C:\Python34\lib\site-packages\piptools\io.py", line 94, in setup
    os.unlink(tmp_part_path)
PermissionError: [WinError 32] Der Prozess kann nicht auf die Datei zugreifen, da sie von einem ande
ren Prozess verwendet wird: 'c:\\Users\\Lehmann\\data\\python\\python34\\moistcalc\\tmpmoibd2g8'
```

**Comment by nvie** [comment]
This exception is a Windows-specific exception related to the atomic-renaming of files. Apparently, the file handle is still open and thus the file cannot be renamed on the Windows platform. We could fix it here, but this particular code is vendored from the excellent [boltons](https://github.com/mahmoud/boltons) library, so it should really be addressed there.

A minimal example that should fail for you is:

```console
$ pip install boltons
```

```python
from boltons.fileutils import AtomicSaver
with AtomicSaver('foo.txt') as f:
    f.write('whatever')
```

That should give the same failure for you probably. If so, please [do report this](https://github.com/mahmoud/boltons/issues/new) to the boltons library, and when it's addressed, I'll vendor back in that latest version.

**Comment by MrLeeh** [comment]
@nvie: You' re right. This is a bug in the [boltons](https://github.com/mahmoud/boltons) library. I' ll address it there and be back with any feedback later on.

**Comment by MrLeeh** [comment]
Allright. Just made contact to the author of boltons. The issue was known and has been fixed but not yet rolled out on PyPi.

**Comment by nvie** [comment]
@MrLeeh Thanks! I've updated the code to match the latest version from boltons. Can you confirm it works for you if you try out the latest `master`? If so, I'll release this to PyPI.

**Comment by MrLeeh** [comment]
Great work. Now it compiles :+1:

**Comment by nvie** [comment]
Awesome! It's now released at 1.1.3 on PyPI.
